tests/lib/ocpp/v201/test_smart_charging_handler.cpp

```diff
@@ -343,6 +362,70 @@ TEST_F(ChargepointTestFixtureV201, K01FR35_IfChargingSchedulePeriodsAreNotInChon
     EXPECT_THAT(sut, testing::Eq(ProfileValidationResultEnum::ChargingSchedulePeriodsOutOfOrder));
 }
 
+TEST_F(ChargepointTestFixtureV201, K01_ValidateChargingStationMaxProfile_NotChargingStationMaxProfile_Invalid) {
+    create_evse_with_id(STATION_WIDE_ID);
+    auto periods = create_charging_schedule_periods({0, 2, 1});
+    auto profile = create_charging_profile(DEFAULT_PROFILE_ID + 1, ChargingProfilePurposeEnum::TxDefaultProfile,
+                                           create_charge_schedule(ChargingRateUnitEnum::A), uuid(),
+                                           ChargingProfileKindEnum::Absolute, DEFAULT_STACK_LEVEL);
+
+    auto sut = handler.validate_charge_point_max_profile(profile, *evses[STATION_WIDE_ID]);
+
+    EXPECT_THAT(sut, testing::Eq(ProfileValidationResultEnum::InvalidProfileType));
+}
+
+TEST_F(ChargepointTestFixtureV201, K04FR03_ValidateChargingStationMaxProfile_EvseIDgt0_Invalid) {
+    create_evse_with_id(DEFAULT_EVSE_ID);
+    std::string same_transaction_id = uuid();
+    open_evse_transaction(DEFAULT_EVSE_ID, same_transaction_id);
+    auto periods = create_charging_schedule_periods({0, 2, 1});
```

Comment:
This is an invalid order of charging schedule periods. They should be in increasing order.

Reply:
Good catch!